clientv3/integration/kv_test.go

```diff
@@ -179,3 +179,104 @@ func TestKVRange(t *testing.T) {
 		}
 	}
 }
+
+func TestKVDeleteRange(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	kv := clientv3.NewKV(clus.RandClient())
+
+	keySet := []string{"a", "b", "c", "c", "c", "d", "e", "f"}
+	for i, key := range keySet {
+		if _, err := kv.Put(key, "", lease.NoLease); err != nil {
+			t.Fatalf("#%d: couldn't put %q (%v)", i, key, err)
+		}
+	}
+
+	tests := []struct {
+		key, end string
+		delRev   int64
+	}{
+		{"a", "b", int64(len(keySet) + 2)}, // delete [a, b)
+		{"d", "f", int64(len(keySet) + 3)}, // delete [d, f)
+	}
+
+	for i, tt := range tests {
+		dresp, err := kv.DeleteRange(tt.key, tt.end)
+		if err != nil {
+			t.Fatalf("#%d: couldn't delete range (%v)", i, err)
+		}
+		if dresp.Header.Revision != tt.delRev {
+			t.Fatalf("#%d: dresp.Header.Revision got %d, want %d", i, dresp.Header.Revision, tt.delRev)
+		}
+		grespKey, err := kv.Get(tt.key, 0)
+		if err != nil {
+			t.Fatalf("#%d: couldn't get key (%v)", i, err)
+		}
+		if len(grespKey.Kvs) > 0 {
+			t.Fatalf("#%d: grespKey.Kvs expected none, but got %+v", i, grespKey.Kvs)
+		}
+		grespEnd, err := kv.Get(tt.end, 0)
+		if err != nil {
+			t.Fatalf("#%d: couldn't get key (%v)", i, err)
+		}
+		if len(grespEnd.Kvs) == 0 {
+			t.Fatalf("#%d: grespEnd.Kvs expected event %s, but got %+v", i, tt.end, grespEnd.Kvs)
+		}
+	}
+}
+
+func TestKVDelete(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	lapi := clientv3.NewLease(clus.RandClient())
+	defer lapi.Close()
+
+	kv := clientv3.NewKV(clus.RandClient())
+
+	resp, err := lapi.Create(context.Background(), 10)
+	if err != nil {
+		t.Fatalf("failed to create lease %v", err)
+	}
+
+	tests := []struct {
+		key, val       string
+		leaseID        lease.LeaseID
+		putRev, delRev int64
+	}{
+		{"foo", "bar", lease.NoLease, 2, 3},
+		{"hello", "world", lease.LeaseID(resp.ID), 4, 5},
```

Comment:
this test case is irrelevant i think. leaseID is not an input of delete. so we probably only need to test one removal case. and it does not need to be table driven.